packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -205,6 +218,22 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 		},
 		async handleHotUpdate(context) {
 			if (context.server.config.isProduction) return;
+			// Check if any of the modules have new content
+			let changes = false;
+			for (const mod of context.modules) {
+				if (changes) break;
+				const path = mod.file ?? fileURLToPath(mod.url.replace('/@fs', 'file:/'));
+				if (!mod.meta) {
+					mod.meta = {};
+				}
+				mod.meta.checksum = await checksum(path)
+				if (checksums.get(path) !== mod.meta.checksum) {
+					changes = true
+					checksums.set(path, mod.meta.checksum)
+				}
+			}
+			// If all the checksums match, skip sending an HMR update
+			if (!changes) return;
```

Comment:
This is a non-logging change. HMR was being triggered on every save. This compares file checksums to determine if we actually need to send an HMR update.

Reply:
Great call, removed this for now! Can revisit in the future if it's a problem.